web-app/src/pages/carbonEmissionPage/view/RideFormView/RideFormView.tsx

```diff
@@ -9,35 +10,124 @@ import {
+`;
+
+const GET_TRANSPORTATIONS = gql`
+  query GetTransportations {
+    transportations {
+      label
+      id
+      carboneEmission
+    }
+  }
+`
 
 export default function RideFormView() {
+  const [formData, setFormData] = useState<CreateRideFormMutationVariables>({
+    label: '',
+    distance: 0,
+    date: '',
+    transportationId: 1,
+  });
+
+  const { data } = useQuery<GetTransportationsQuery>(GET_TRANSPORTATIONS)
+
+  const updateFormData = (partialFormData: Partial<CreateRideFormMutationVariables>) => {
+    setFormData({ ...formData, ...partialFormData });
+  };
+
+  const [createRideMutation] = useMutation<CreateRideFormMutation, CreateRideFormMutationVariables>(CREATE_RIDE);
+
+  const createRide = async () => {
+    await createRideMutation({
+      variables: {
+        label: formData.label,
+        distance: formData.distance,
+        date: formData.date,
+        transportationId: formData.transportationId,
+      },
+    });
+  };
+
   return (
     <>
       <FormViewStyled>
         <FormTitle>Nouveau trajet :</FormTitle>
-        <Form>
+        <Form
+          aria-label='form'
+          onSubmit={(event) => {
+            event.preventDefault();
+            console.log(formData)
+            createRide();
+          }}
+        >
           <FormLabelWithField>
             Nom du trajet:
-            <FormTextField type='text' />
+            <FormTextField
+              type='text'
+              required
+              minLength={2}
+              onChange={(event) => {
+                updateFormData({ label: event.target.value });
+              }}
+            />
           </FormLabelWithField>
           <FormLabelWithField>
             Distance en km:
-            <FormTextField type='text' />
+            <FormTextField
+              type='number'
+              required
+              onChange={(event) => {
+                updateFormData({ distance: parseInt(event.target.value) });
+              }}
+            />
           </FormLabelWithField>
           <FormLabelWithField>
             Date:
-            <FormTextField type='date' />
+            <FormTextField
+              type='date'
+              required
+              onChange={(event) => {
+                updateFormData({ date: new Date(event.target.value).toISOString() });
+              }}
+            />
           </FormLabelWithField>
           <FormLabelWithField>
             Moyen de transport:
-            <FormSelect>
-              <option value=''>Select</option>
-              <option value='car'>Voiture</option>
-              <option value='bike'>Bus</option>
-              <option value='publicTransport'>Avion</option>
+            <FormSelect
+              required
+              onChange={(event) => {
+                updateFormData({
+                  transportationId: parseInt(event.target.value),
+                });
+              }}
+            >
+              <option value=''>Sélectionner</option>
```

Comment:
Cette option est-elle nécessaire ? Elle sert de placeholder certes, mais ce n'est pas nécessaire vu qu'il y a déjà un label, et elle risque de causer une erreur si elle est sélectionnée par l'utilisateur et soumise.

Reply:
c'est noté merci Arnaud